test/inspect-test.js

```diff
@@ -4,368 +4,554 @@ const tape = require('tape');
-      '  01, 40, 60, 80, f0, ff\n' +
-      ']>');
-    t.ok(
-        re.test(lines),
-        'typed array content');
-    sess.send(`v8 inspect --array-length 1 ${uint8Array}`);
-  });
-
-  sess.linesUntil(/]>/, (err, lines) => {
-    t.error(err);
-    lines = lines.join('\n');
-    const re = new RegExp(
-      '0x[0-9a-f]+:' +
-      '<ArrayBufferView: backingStore=0x[0-9a-f]+, byteOffset=\\d+, ' +
-      'byteLength=6: \\[\n' +
-      '  01 ...\n' +
-      ']>');
-    t.ok(
-        re.test(lines),
-        'typed array content with maximum length 1');
-    sess.send(`v8 inspect ${buffer}`);
-  });
-
-  sess.linesUntil(/]>/, (err, lines) => {
-    t.error(err);
-    lines = lines.join('\n');
-    const re = new RegExp(
-      '0x[0-9a-f]+:' +
-      '<ArrayBufferView: backingStore=0x[0-9a-f]+, byteOffset=\\d+, ' +
-      'byteLength=6: \\[\n' +
-      '  ff, f0, 80, 0f, 01, 00\n' +
-      ']>');
-    t.ok(
-        re.test(lines),
-        'buffer content');
-    sess.send(`v8 inspect --array-length 1 ${buffer}`);
-  });
-
-  sess.linesUntil(/]>/, (err, lines) => {
-    t.error(err);
-    lines = lines.join('\n');
-    const re = new RegExp(
-      '0x[0-9a-f]+:' +
-      '<ArrayBufferView: backingStore=0x[0-9a-f]+, byteOffset=\\d+, ' +
-      'byteLength=6: \\[\n' +
-      '  ff ...\n' +
-      ']>');
-    t.ok(
-        re.test(lines),
-        'buffer content with maximum length 1');
-    sess.send(`v8 inspect -s ${arrowFunc}`);
-  });
-
-
-  sess.linesUntil(/^>/, (err, lines) => {
-    t.error(err);
-    lines = lines.join('\n');
-    // Include 'source:' and '>' to act as boundaries. (Avoid
-    // passing if the whole file it displayed instead of just
-    // the function we want.)
-    const arrowSource = 'source:\n' +
-        'function c.hashmap.(anonymous function)(a,b)=>{a+b}\n' +
-        '>';
-
-    t.ok(lines.includes(
-        arrowSource),
-        'arrow method source');
-
-    sess.send(`v8 inspect -s ${fn}`);
+    if (err) {
+      return teardown(t, sess, err);
+    }
+    const parent = 'hashmap';
+    const addresses = collectMembers(
+        t, lines.join('\n'), hashMapTests, parent);
+    verifyMembers(t, sess, addresses, hashMapTests, parent, teardown);
   });
+}
+
+function teardown(t, sess, err) {
+  t.end(err);
+  sess.quit();
+}
+
+function collectMembers(t, lines, tests, parent) {
+  const addresses = {};
+  for (let name of Object.keys(tests)) {
+    const test = tests[name];
+    const match = lines.match(test.re);
+    if (match) {
+      t.pass(`${parent}${test.desc} should exist `);
+      addresses[name] = match[1];
+    } else {
+      if (test.optional) {
+        t.skip(`${test.desc}: ${test.optional}`);
+      } else {
+        t.fail(`${parent}${test.desc} should exist `);
+      }
+    }
+  }
+  return addresses;
+}
 
-  sess.linesUntil(/^>/, (err, lines) => {
-    t.error(err);
-    lines = lines.join('\n');
+// To avoid lldb output being mixed, we must do the validation one by one
```

Comment:
What does this mean, mixed?

Reply:
I think if simply we loop over the validators, it is possible that one `sess.linesUntil()` would consume the output of `sess.send()` from another validator (if one of them fails and receives the output from another that accidentally contains the marker that it's waiting for, the test results could be pretty bizzare).